src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2701,8 +2701,11 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 	{
 		InsertValues *insertValues = (InsertValues *) lfirst(insertValuesCell);
 		List *prunedShardIntervalList = NIL;
-		Expr *partitionValueExpr = (Expr *) strip_implicit_coercions(
-			(Node *) insertValues->partitionValueExpr);
+		Expr *partitionValueExpr = (Expr *) eval_const_expressions(NULL,
+																   strip_implicit_coercions(
```

Comment:
maybe these calls should be swapped, i.e
```suggestion
		Expr *partitionValueExpr = (Expr *) strip_implicit_coercions(NULL,
																   eval_const_expressions(
```